app/lib/experience.js

```diff
@@ -0,0 +1,124 @@
+//functions intereacting with player experience points in Minecraft will go here.
+module.exports = Base => class extends Base {
+    constructor() {
+        super();
+    }
+
+    //minecraft seperates levels and points you need to get both.
+    readPlayerExperiencePoints(playerName) {
+        return new Promise((resolve) => {
+            const listenForData = (data) => {
+                let text = data.toString();
+                let regEx = new RegExp(`.*\\s${playerName}\\shas\\s\\d+\\sexperience\\spoints`);
+
+                if (!regEx.test(text)) return;
+                this.serverProcess.stdout.removeListener('data', listenForData);
+
+                let points = parseInt(text.split(' ')[4],10);
+                resolve(points);
+            };
+
+            this.serverProcess.stdout.on('data', listenForData);
+            this.writeToMine(`experience query ${playerName} points`);
+        });
+    }
+
+    //minecraft seperates levels and points you need to get both.
+    readPlayerExperienceLevels(playerName) {
+        return new Promise((resolve) => {
+            const listenForData = (data) => {
+                let text = data.toString();
+                let regEx = new RegExp(`.*\\s${playerName}\\shas\\s\\d+\\sexperience\\slevels`);
+
+                if (!regEx.test(text)) return;
+                this.serverProcess.stdout.removeListener('data', listenForData);
+
+                let levels = parseInt(text.split(' ')[4],10);
+                console.log(levels);
+                resolve(levels);
+            };
+
+            this.serverProcess.stdout.on('data', listenForData);
+            this.writeToMine(`experience query ${playerName} levels`);
+        });
+    }
+
+    async addPlayerExperience(playerName, newExp) {
+        //need to convert existing level/points into points, then add to that pointPool and convert the new total back.
+        let currentExp = this.convertLevelsToPoints(await this.readPlayerExperienceLevels(playerName),await this.readPlayerExperiencePoints(playerName));
+        let totalExp = parseInt(newExp,10) + parseInt(currentExp,10);
```

Comment:
You shouldn't have to worry about NaN when you parse something from the minecraft server. or if you've already checked. You can just resolve this Just didn't wanna send you down a path here like previous comments